cid.go

```diff
@@ -168,15 +174,29 @@ func NewPrefixV1(codecType uint64, mhType uint64) Prefix {
 // Cid represents a self-describing content adressed
 // identifier. It is formed by a Version, a Codec (which indicates
 // a multicodec-packed content type) and a Multihash.
-type Cid struct {
-	version uint64
-	codec   uint64
-	hash    mh.Multihash
+// Byte layout: [version, codec, multihash]
+// - version uvarint
+// - codec uvarint
+// - hash mh.Multihash
+type Cid string
+
+var EmptyCid = Cid(string([]byte{}))
```

Comment:
Why `string([]byte{})` and not just `""`.

Reply:
I would also make this a constant so we have:
```
const EmptyCid = Cid("")
```

Reply:
Also. I find `EmptyCid` a bit tedious of a name.  How out just `Ni`"?  When used outside this package it would be `cid.Nil`.